src/registrar/tests/test_transition_domain_migrations.py

```diff
@@ -21,6 +21,112 @@
 from .common import MockEppLib, less_console_noise
 
 
+class TestPatchAgencyInfo(TestCase):
+    def setUp(self):
+        self.user, _ = User.objects.get_or_create(username="testuser")
+        self.domain, _ = Domain.objects.get_or_create(name="testdomain.gov")
+        self.domain_info, _ = DomainInformation.objects.get_or_create(domain=self.domain, creator=self.user)
+        self.transition_domain, _ = TransitionDomain.objects.get_or_create(
+            domain_name="testdomain.gov", federal_agency="test agency"
+        )
+
+    def tearDown(self):
+        Domain.objects.all().delete()
+        DomainInformation.objects.all().delete()
+        User.objects.all().delete()
+        TransitionDomain.objects.all().delete()
+
+    @patch("registrar.management.commands.utility.terminal_helper.TerminalHelper.query_yes_no_exit", return_value=True)
+    def call_patch_federal_agency_info(self, mock_prompt):
+        """Calls the patch_federal_agency_info command and mimics a keypress"""
+        call_command("patch_federal_agency_info", "registrar/tests/data/fake_current_full.csv", debug=True)
+
+    def test_patch_agency_info(self):
+        """
+        Tests that the `patch_federal_agency_info` command successfully
+        updates the `federal_agency` field
+        of a `DomainInformation` object when the corresponding
+        `TransitionDomain` object has a valid `federal_agency`.
+        """
+        self.call_patch_federal_agency_info()
+
```

Comment:
Assert that the domain_info.federal_agency is NOT "test agency" before patch is called